_posts/2022-01-10-reassign-partitions.md

```diff
@@ -0,0 +1,416 @@
+Let's start the shell process:
+
+```sh
+kubectl exec -ti <INTERACTIVE-POD-NAME> /bin/bash
+```
+
+Now it's time to generate the `reassignment.json` file. 
+We will use the `kafka-reassign-partitions.sh` command to generate our proposal `reassignment.json` data:
+
+```sh
+bin/kafka-reassign-partitions.sh --bootstrap-server <CLUSTER-NAME>-kafka-bootstrap:9092 \
+--topics-to-move-json-file /tmp/topics.json \
+--broker-list 0,1,2 \
+--generate
+```
+Here `topics-to-move-json-file` points towards the `topics.json file` and `--broker-list` is the list of broker which will now handle the partitions which are present with the broker to be scaled down.
+
+Once you run this command, you will be able to see the JSON data which is generated by the Kafka reassignment partition tool. You get the current replica assignment and the proposed `reassignment.json` data.
+
+```sh
+Current partition replica assignment
+{"version":1,"partitions":[{"topic":"my-topic-two","partition":0,"replicas":[0,3,1],"log_dirs":["any","any","any"]},{"topic":"my-topic-two","partition":1,"replicas":[1,0,4],"log_dirs":["any","any","any"]},{"topic":"my-topic-two","partition":2,"replicas":[4,1,2],"log_dirs":["any","any","any"]},{"topic":"my-topic-two","partition":3,"replicas":[2,4,3],"log_dirs":["any","any","any"]},{"topic":"my-topic-two","partition":4,"replicas":[3,2,0],"log_dirs":["any","any","any"]},{"topic":"my-topic-two","partition":5,"replicas":[0,1,4],"log_dirs":["any","any","any"]},{"topic":"my-topic-two","partition":6,"replicas":[1,4,2],"log_dirs":["any","any","any"]},{"topic":"my-topic-two","partition":7,"replicas":[4,2,3],"log_dirs":["any","any","any"]},{"topic":"my-topic-two","partition":8,"replicas":[2,3,0],"log_dirs":["any","any","any"]},{"topic":"my-topic-two","partition":9,"replicas":[3,0,1],"log_dirs":["any","any","any"]}]}
+
+Proposed partition reassignment configuration
+{"version":1,"partitions":[{"topic":"my-topic-two","partition":0,"replicas":[2,0,1],"log_dirs":["any","any","any"]},{"topic":"my-topic-two","partition":1,"replicas":[0,1,2],"log_dirs":["any","any","any"]},{"topic":"my-topic-two","partition":2,"replicas":[1,2,0],"log_dirs":["any","any","any"]},{"topic":"my-topic-two","partition":3,"replicas":[2,1,0],"log_dirs":["any","any","any"]},{"topic":"my-topic-two","partition":4,"replicas":[0,2,1],"log_dirs":["any","any","any"]},{"topic":"my-topic-two","partition":5,"replicas":[1,0,2],"log_dirs":["any","any","any"]},{"topic":"my-topic-two","partition":6,"replicas":[2,0,1],"log_dirs":["any","any","any"]},{"topic":"my-topic-two","partition":7,"replicas":[0,1,2],"log_dirs":["any","any","any"]},{"topic":"my-topic-two","partition":8,"replicas":[1,2,0],"log_dirs":["any","any","any"]},{"topic":"my-topic-two","partition":9,"replicas":[2,1,0],"log_dirs":["any","any","any"]}]}
+
+```
+Now you can copy this proposed `reassignment.json` data and paste it in a `reassignment.json` file, which will be copied to our interactive pod container in the next step.
```

Comment:
`this proposed `reassignment.json` data` -> `this proposed  partition reassignment`

Reply:
Do we need `, which will be copied to our interactive pod container in the next step.` here? We repeat it again just below